gammapy/spectrum/fit.py

```diff
@@ -3,15 +3,105 @@
 import copy
 import numpy as np
 import astropy.units as u
-from ..utils.fitting import Fit
+from ..utils.fitting import Fit, Parameters
 from .. import stats
 from .utils import CountsPredictor
 from .observation import SpectrumObservationList, SpectrumObservation
 
-__all__ = ["SpectrumFit"]
+__all__ = ["SpectrumFit", "SpectrumDataset"]
 
 log = logging.getLogger(__name__)
 
+class SpectrumDataset:
+    """Perform spectral model likelihood fit on a CountsSpectrum.
```

Comment:
The docstring confuses me. Is it really performing a fit?

Reply:
No, not really. It contains the data and a model and computes the resulting likelihood. It is used as argument by `Fit`.
This docstring is a copy of the docstring used so far for `Dataset`. We will correct all at once.

Reply:
I can do this we renaming the `likelihood()` methods for all the `Dataset` classes